modules/standard/IO.chpl

```diff
@@ -2661,9 +2656,8 @@ record ioChar {
 pragma "no doc"
 inline proc _cast(type t, x: ioChar) where t == string {
   var csc: c_string_copy =  qio_encode_to_string(x.ch);
-  var len = csc.length;
   // The caller has responsibility for freeing the returned string.
-  return new string(csc:c_ptr(uint(8)), len, len, owned=true);
+  return new string(csc);
```

Comment:
This seems the same as the other c_string_copy situations, but here it's copying..
I'm beginning to think that the best thing would be to create a function or constructor for a string that takes in a c_string_copy by ref and "steals" it, setting the argument to nil. I think that generally there's only one thing we want to happen in these cases, and that the current approach of relying on conversion to c_string and explicit needToCopy and owned arguments is (unnecessarily) hard to get right.